packages/core/src/utils/json.ts

```diff
@@ -1,5 +1,10 @@
+import { type JsonObject, jsonObjectGuard } from '@logto/schemas';
 import { trySafe } from '@silverhand/essentials';
 
 export const safeParseJson = (jsonString: string): unknown =>
   // eslint-disable-next-line no-restricted-syntax
   trySafe(() => JSON.parse(jsonString) as unknown);
+
+// Safely parse Zod unknown JSON object to JsonObject
+export const safeParseUnknownJson = (unknownJson: Record<string, unknown>): JsonObject =>
+  trySafe(() => jsonObjectGuard.safeParse(unknownJson).data) ?? {};
```

Comment:
By looking into the definition of `jsonObjectGuard`, I'm curious what could happen if there are some explicit undefined fields in unknownJson or it's subfields. Could you please double check?

Reply:
updated, deep clean the undefined keys.